### ISSUE TITLE
Non-deterministic behavior when raising some exceptions

### ISSUE THREAD

**Issue by pawelmhm** [Nykakin/chompjs#30]
I'm using chompjs in my project, and I have some unit tests to identify some potential problems, I found out that there is some unusual behavior related to exception handling. Let's use following trivial test:

```python
import chompjs
import pytest
import re


def test_chompjs_html_quoted_quotations_decoded():
    text = '{"test": """}'
    with pytest.raises(ValueError) as e:
        chompjs.parse_js_object(text)

    assert re.search('Error parsing', str(e.value))

```

When I run this test I get several good tests results, majority of times it passes, but once in a while it fails. See following screenshot. When it fails it fails with following message:

```
>       assert re.search('Error parsing', str(e.value))
E       assert None
E        +  where None = <function search at 0x7f5f8fc1d870>('Error parsing', "'utf-8' codec can't decode byte 0xe0 in position 39: invalid continuation byte")
E        +    where <function search at 0x7f5f8fc1d870> = re.search
E        +    and   "'utf-8' codec can't decode byte 0xe0 in position 39: invalid continuation byte" = str(UnicodeDecodeError('utf-8', b'{"test":"""}\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00...x00\x00\xa6\xda\xcc\xd9U\x00\x00\x0b\x00\x00\x00\x00\x00\x00\x00\x11""\xe9\xfc"}', 39, 40, 'invalid continuation byte'))
E        +      where UnicodeDecodeError('utf-8', b'{"test":"""}\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00...x00\x00\xa6\xda\xcc\xd9U\x00\x00\x0b\x00\x00\x00\x00\x00\x00\x00\x11""\xe9\xfc"}', 39, 40, 'invalid continuation byte') = <ExceptionInfo UnicodeDecodeError('utf-8', b'{"test":"""}\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\...a6\xda\xcc\xd9U\x00\x00\x0b\x00\x00\x00\x00\x00\x00\x00\x11""\xe9\xfc"}', 39, 40, 'invalid continuation byte') tblen=2>.value

```

See test runs here

![chompjs](https://user-images.githubusercontent.com/2700942/144425338-7f0cc899-d898-474f-be8d-3fae47761515.png)

I added breakpoint to see this in detail and this exception value looks like this

```
 /home/pawel/PycharmProjects/playground/something.py(14)test_chompjs_html_quoted_quotations_decoded()->None
-> breakpoint()
(Pdb) e.value
UnicodeDecodeError('utf-8', b'{"test":"""}\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00W\x00\x00\x00\x00\x00\x00\x00pS\x98)=\x7f\x00\x00\x00\xc0A*=\x7f\x00\x000\xf8\xe0y\xa4U\x00\x00\n\x00\x00\x00\n\x00\x00\x00p?\x00\x00P?\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00@\xb4\xe2y\xa4U\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x02\x00\x00\x00C\x00\x00\x00\x92\x00\x00\x00=\x7f\x00\x00\x90\'\x95)=\x7f\x00\x00\xf0%\x95)=\x7f\x00\x00@\x1f\x94)=\x7f\x00\x00 &\x95)=\x7f\x00\x00p\x00n*=\x7f\x00\x00p\x00n*=\x7f\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xf0V\x95)=\x7f\x00\x00\xf0En*=\x7f\x00\x00\xc0\'\x95)=\x7f\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x7f\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00@\xb4\xe2y\xa4U\x00\x00\x04\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x06\x00\x00\x00\x06\x00\x00\x00C\x00\x00\x00\x95\x00\x00\x00=\x7f\x00\x000\x89\x97)=\x7f\x00\x00\x80\x15\x94)=\x7f\x00\x00\xe0<\x96)=\x7f\x00\x00\xa0\x89\x97)=\x7f\x00\x00p\x00n*=\x7f\x00\x00p\x00n*=\x7f\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xf0V\x95)=\x7f\x00\x00\xb0Bn*=\x7f\x00\x00\xf0\'\x95)=\x7f\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00\x08\x00\x00\x00\x00\x00\x00\x00\x80\x12\xacy\xa4U\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x05\x00\x00\x00\x00\x00\x00\x00\xff\xff\xff\x00\x04\x01\x02\x03\xb3\x9fP3\x84En\x8cp\x14\x19*=\x7f\x00\x00@0\xe0y\xa4U\x00\x00\xa5b\x06C\xd9\xc5\x19\xddpU\xfc)=\x7f\x00\x00@0\xe0y\xa4U\x00\x00\x96\x9a\x0e!\xe8\xeeIL\xf0\xb2\xfb)=\x7f\x00\x00@0\xe0y\xa4U\x00\x00\x07\xee\xc8\xc1\xec\xad\xdb}\xb0\x14\x19*=\x7f\x00\x00@0\xe0y\xa4U\x00\x00\x8b\xc4\x88\x1d\xc3*c^\xb0\xa5\xfb)=\x7f\x00\x00\xa0\xf2\xe0y\xa4U\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00@\xb4\xe2y\xa4U\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x02\x00\x00\x00C\x00\x00\x00\xa3\x00\x00\x00\x00\x00\x00\x00P(\x95)=\x7f\x00\x00\xf0%\x95)=\x7f\x00\x00\x80\x1d\x94)=\x7f\x00\x00 &\x95)=\x7f\x00\x00p\x00n*=\x7f\x00\x00p\x00n*=\x7f\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xf0V\x95)=\x7f\x00\x00\xf0En*=\x7f\x00\x00\x80(\x95)=\x7f\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00@\xb4\xe2y\xa4U\x00\x00\x04\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x04\x00\x00\x00\x05\x00\x00\x00C\x00\x00\x00\xa6\x00\x00\x00\x00\x00\x00\x000\x0e\x98)=\x7f\x00\x00\x80\x12\x94)=\x7f\x00\x00\x80\x1a\x94)=\x7f\x00\x00\x90\x80\x98)=\x7f\x00\x00p\x00n*=\x7f\x00\x00p\x00n*=\x7f\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xf0V\x95)=\x7f\x00\x00\xb0Bn*=\x7f\x00\x00\xb0(\x95)=\x7f\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00\x08\x00\x00\x00\x00\x00\x00\x00\x80\x12\xacy\xa4U\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x05\x00\x00\x00\x00\x00\x00\x00\xff\xff\xff\x00\x04\x01\x02\x03\xb3\x9fP3\x84En\x8cp\x14\x19*=\x7f\x00\x00@0\xe0y\xa4U\x00\x00\xa5b\x06C\xd9\xc5\x19\xddpU\xfc)=\x7f\x00\x00@0\xe0y\xa4U\x00\x00\x96\x9a\x0e!\xe8\xeeIL\xf0\xb2\xfb)=\x7f\x00\x00@0\xe0y\xa4U\x00\x00\x07\xee\xc8\xc1\xec\xad\xdb}\xb0\x14\x19*=\x7f\x00\x00@0\xe0y\xa4U\x00\x00\x8b\xc4\x88\x1d\xc3*c^\xb0\xa5\xfb)=\x7f\x00\x00\xa0\xf2\xe0y\xa4U\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00\x08\x00\x00\x00\x00\x00\x00\x00\x80\x12\xacy\xa4U\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x05\x00\x00\x00\x00\x00\x00\x00\xff\xff\xff\x00\x04\x01\x02\x03\xb3\x9fP3\x84En\x8cp\x14\x19*=\x7f\x00\x00@0\xe0y\xa4U\x00\x00\xa5b\x06C\xd9\xc5\x19\xddpU\xfc)=\x7f\x00\x00@0\xe0y\xa4U\x00\x00\x96\x9a\x0e!\xe8\xeeIL\xf0\xb2\xfb)=\x7f\x00\x00@0\xe0y\xa4U\x00\x00\x07\xee\xc8\xc1\xec\xad\xdb}\xb0\x14\x19*=\x7f\x00\x00@0\xe0y\xa4U\x00\x00\x8b\xc4\x88\x1d\xc3*c^\xb0\xa5\xfb)=\x7f\x00\x00\xa0\xf2\xe0y\xa4U\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00\x08\x00\x00\x00\x00\x00\x00\x00\xe0\x14\xacy\xa4U\x00\x00\x05\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xff\xff\xff\xff\xff\xff\xff\xff\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00\x08\x00\x00\x00\x00\x00\x00\x00\xe0\x14\xacy\xa4U\x00\x00\x05\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xff\xff\xff\xff\xff\xff\xff\xff\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00\x08\x00\x00\x00\x00\x00\x00\x00\x80\x12\xacy\xa4U\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x05\x00\x00\x00\x00\x00\x00\x00\xff\xff\xff\x00\x04\x01\x02\x03\xb3\x9fP3\x84En\x8cp\x14\x19*=\x7f\x00\x00@0\xe0y\xa4U\x00\x00\xa5b\x06C\xd9\xc5\x19\xddpU\xfc)=\x7f\x00\x00@0\xe0y\xa4U\x00\x00\x96\x9a\x0e!\xe8\xeeIL\xf0\xb2\xfb)=\x7f\x00\x00@0\xe0y\xa4U\x00\x00\x07\xee\xc8\xc1\xec\xad\xdb}\xb0\x14\x19*=\x7f\x00\x00@0\xe0y\xa4U\x00\x00\x8b\xc4\x88\x1d\xc3*c^\xb0\xa5\xfb)=\x7f\x00\x00\xa0\xf2\xe0y\xa4U\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00\x08\x00\x00\x00\x00\x00\x00\x00\x80\x12\xacy\xa4U\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x05\x00\x00\x00\x00\x00\x00\x00\xff\xff\xff\x00\x04\x01\x02\x03\xb3\x9fP3\x84En\x8cp\x14\x19*=\x7f\x00\x00@0\xe0y\xa4U\x00\x00\xa5b\x06C\xd9\xc5\x19\xddpU\xfc)=\x7f\x00\x00@0\xe0y\xa4U\x00\x00\x96\x9a\x0e!\xe8\xeeIL\xf0\xb2\xfb)=\x7f\x00\x00@0\xe0y\xa4U\x00\x00\x07\xee\xc8\xc1\xec\xad\xdb}\xb0\x14\x19*=\x7f\x00\x00@0\xe0y\xa4U\x00\x00\x8b\xc4\x88\x1d\xc3*c^\xb0\xa5\xfb)=\x7f\x00\x00\xa0\xf2\xe0y\xa4U\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00\x08\x00\x00\x00\x00\x00\x00\x00\x80\x12\xacy\xa4U\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x05\x00\x00\x00\x00\x00\x00\x00\xff\xff\xff\x00\x04\x01\x02\x03\xb3\x9fP3\x84En\x8cp\x14\x19*=\x7f\x00\x00@0\xe0y\xa4U\x00\x00\xa5b\x06C\xd9\xc5\x19\xddpU\xfc)=\x7f\x00\x00@0\xe0y\xa4U\x00\x00\x96\x9a\x0e!\xe8\xeeIL\xf0\xb2\xfb)=\x7f\x00\x00@0\xe0y\xa4U\x00\x00\x07\xee\xc8\xc1\xec\xad\xdb}\xb0\x14\x19*=\x7f\x00\x00@0\xe0y\xa4U\x00\x00\x8b\xc4\x88\x1d\xc3*c^\xb0\xa5\xfb)=\x7f\x00\x00\xa0\xf2\xe0y\xa4U\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00\x08\x00\x00\x00\x00\x00\x00\x00\x80\x12\xacy\xa4U\x00\x00\x03\x00\x00\x00\x00\x00\x00\x00\x02\x00\x00\x00\x00\x00\x00\x00\xff\xff\x01\x00\xff\xff\xff\xff#__\xce\xaa\xb5{*\xb0\x13\x19*=\x7f\x00\x00p\x02N*=\x7f\x00\x00""\x04_"}', 41, 42, 'invalid start byte')
(Pdb) e
<ExceptionInfo UnicodeDecodeError('utf-8', b'{"test":"""}\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00...xff\xff\xff\xff#__\xce\xaa\xb5{*\xb0\x13\x19*=\x7f\x00\x00p\x02N*=\x7f\x00\x00""\x04_"}', 41, 42, 'invalid start byte') tblen=2>
(Pdb) 

```

I have this test there for a long time, and it only breaks with new version, perhaps this https://github.com/Nykakin/chompjs/commit/63b303ddae910be1cc86864952687d4d288a643c introduced some regression @Nykakin

**Comment by Nykakin** [comment]
Should be resolved by https://github.com/Nykakin/chompjs/pull/31 included in 1.1.5 release.

**Comment by pawelmhm** [comment]
yes works fine
